boot/kernel_os.go

```diff
@@ -155,7 +166,7 @@ func SetNextBoot(s *snap.Info) error {
 
 // KernelOrOsRebootRequired returns whether a reboot is required to swith to the given OS or kernel snap.
 func KernelOrOsRebootRequired(s *snap.Info) bool {
-	if s.Type != snap.TypeKernel && s.Type != snap.TypeOS {
+	if s.Type != snap.TypeKernel && s.Type != snap.TypeOS && s.Type != snap.TypeBase {
```

Comment:
this is correct for unrelated bases because of the check on nextBoot, feels still a bit fragile, either way the unrelated  base case merits new/own tests I think